release/models/platform/openconfig-platform-healthz-fault.yang

```diff
@@ -0,0 +1,293 @@
+      }
+    }
+  }
+
+  grouping fault-state {
+    description
+      "The fault's operational state";
+    leaf symptom {
+      type identityref {
+        base SYMPTOM_BASE;
+      }
+      description
+        "The symptom that raised this fault";
+    }
+
+    leaf origin-time {
+      type oc-types:timeticks64;
+      description
+        "The time at which the fault first became active.
+        This value should not change.";
+    }
+
+    leaf last-detection-time {
+      type oc-types:timeticks64;
+      description
+        "The time at which the fault's underlying condition was last sampled
+        and confirmed asserted.";
+    }
+
+    leaf occurrences {
+      type uint64;
+      description
+        "The number of times this fault transitioned into the active state.";
+    }
+
+    leaf description {
+      type string;
+      description
+        "Description of the fault to assist with debuggability";
+    }
+
+    leaf status {
+      type enumeration {
+        enum UNSPECIFIED {
+          description
+            "The component's health status has not yet been checked
+            by the system.";
+        }
+
+        enum ACTIVE {
+          description
+            "The underlying condition is asserted";
+        }
+
+        enum INACTIVE {
+          description
+            "The underlying condition is not asserted.";
+        }
+      }
+      description
+        "The status of the fault, indicates whether the underlying condition
+        is actively asserted.";
+    }
+
+    leaf severity {
+      type identityref {
+        base oc-alarm-types:OPENCONFIG_ALARM_SEVERITY;
+      }
+      description
+        "The type of alarm to which the thresholds apply.";
+    }
+  }
+
+  grouping fault-remediations {
+    description
+      "Grouping containing a faults recommended remediations";
+
+    container remediations {
+      description
+        "Enclosing container for a fault's recommended remediations.";
+
+      list remediation {
+        key "index";
+        leaf index {
+          type leafref {
+            path "../state/index";
+          }
+          description
+            "This index indicates the order in which the actions should be
+            applied. The order should progress from least to most disruptive.";
+        }
+        description
+          "List of recommended actions to respond to the fault";
+
+        container state {
+          config false;
+          description
+            "The remediation's operational state.";
+          leaf index {
+            type uint64;
+            description
+              "This index indicates the order in which the actions should be
+              applied. The order should progress from least to most
+              disruptive.";
+          }
+
+          leaf action {
+            description
+              "The specific action to be performed";
+            type identityref {
+              base ACTION_BASE;
+            }
+          }
```

Comment:
Should this only be one action? Are there scenarios when multiple actions may be performed to address the fault? e.g. `ACTION_WARM_REBOOT` _or_ `ACTION_COLD_REBOOT`

Alternatively, is there any ordering between actions where performing action A also covers remediations performed by action B?